### PR TITLE
[pkg/telemetryquerylanguage] Unexport unnessessary functions and types

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
This PR unexports several functions and types that do not need to be exposed

**Link to tracking Issue:**
Works towards #13690 

**Testing:** 
Unit Tests

### CODE REVIEW

File: processor/transformprocessor/internal/metrics/func_convert_summary_count_val_to_sum_test.go [hunk 1]
```diff
@@ -17,101 +17,19 @@ package metrics
...
-}
-
-type summaryTestCase struct {
-	name  string
-	input pmetric.Metric
-	inv   tql.Invocation
-	want  func(pmetric.MetricSlice)
-}
-
-func Test_ConvertSummarySumValToSum(t *testing.T) {
```
Comment (TylerHelmuth) [inline processor/transformprocessor/internal/metrics/func_convert_summary_count_val_to_sum_test.go:99]: These 2 test files shouldn't been using the function directly but were still calling NewFunctionCall.  They are now in line with how we test other functions.

File: pkg/telemetryquerylanguage/tql/functions.go [hunk 2]
```diff
@@ -19,12 +19,13 @@ import (
 	"reflect"
 )
 
-type PathExpressionParser func(*Path) (GetSetter, error)
+type pathExpressionParser func(*Path) (GetSetter, error)
 
-type EnumParser func(*EnumSymbol) (*Enum, error)
+type enumParser func(*EnumSymbol) (*Enum, error)
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply pkg/telemetryquerylanguage/tql/functions.go:24]: If it is needed to be provided by the user then needs to be public.

[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/tql/functions.go]: reverted

File: unreleased/issue-13690-adjust-exported-funcs.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: pkg/telemetryquerylanguage
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Unexport `BoolExpressionEvaluator`, `Literal`, `NewGetter`, `PathExpressionParser`, `EnumParser`, and `NewFunctionCall`.
```
Comment (evan-bradley) [inline unreleased/issue-13690-adjust-exported-funcs.yaml:8]: You'll want to remove `PathExpressionParser` and `EnumParser` from this list.

**Review by evan-bradley — COMMENTED** [review]
Since this PR focuses on the TQL package's public interface, I think it may be worth adding a section in the readme to describe how to use the TQL in a component, as it's not immediately obvious when looking through the docs in the repo or [the published docs](https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tql). However, that change isn't essential here and may be large enough in scope to make more sense in another PR.

File: unreleased/issue-13690-adjust-exported-funcs.yaml (hunk 3 above)
Comment (TylerHelmuth) [reply unreleased/issue-13690-adjust-exported-funcs.yaml:8]: You're right, thanks

**Review by evan-bradley — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Need rebase

**Comment by TylerHelmuth** [comment]
@bogdandrutu merge conflicts handled

**Review by djaglowski — APPROVED** [review]

**Pull request merged**